metis-normalization/src/main/java/eu/europeana/normalization/normalizers/DatesNormalizer.java

```diff
@@ -198,10 +198,13 @@ private void normalizeElement(Document document, Element element, Namespace.Elem
 
     // Apply the normalization. If nothing can be done, we return.
     final String elementText = XmlUtil.getElementText(element);
-    final DateNormalizationResult dateNormalizationResult = normalizationFunction.apply(
-        elementText);
-    if (dateNormalizationResult.getDateNormalizationExtractorMatchId()
-        == DateNormalizationExtractorMatchId.NO_MATCH) {
+    // TODO: 11/10/2022 Check if this can be more securely structured.
+    //  Currently the class InstantEdtfDate is allowed to contain invalid values for day, month, year and therefore a call
+    //  to some of it's functions, for example getLastDay which internally triggers a call to lastDayBasedOnMonth can fail if the values are invalid.
+    //  Probably a better approach is to not allow the object to be created with invalid values in the first place.
+    final DateNormalizationResult dateNormalizationResult = normalizationFunction.apply(elementText);
+    if (dateNormalizationResult.getDateNormalizationExtractorMatchId() == DateNormalizationExtractorMatchId.NO_MATCH
+        || dateNormalizationResult.getDateNormalizationExtractorMatchId() == DateNormalizationExtractorMatchId.INVALID) {
```

Comment:
Update `NormalizationTest` Unit Test I tested with the following records.

```
 private static Stream<Arguments> providedRecords() {
    return Stream.of(
        Arguments.of("edm-record-date-1.xml"),
        Arguments.of("edm-record-date-2.xml"),
        Arguments.of("edm-record-date-3.xml"),
        Arguments.of("edm-record-date-4.xml"),
        Arguments.of("edm-record-date-5.xml"),
        Arguments.of("edm-record-internal.xml")
    );
  }
  @ParameterizedTest
  @MethodSource("providedRecords")
  void testDateNormalization(String inputRdf) throws NormalizationException, NormalizationConfigurationException, SerializationException {

    final RdfConversionUtils rdfConversionUtils = new RdfConversionUtils();
    //Verify conversion before normalization
    final RDF rdf = rdfConversionUtils.convertInputStreamToRdf(
        getClass().getClassLoader().getResourceAsStream(inputRdf));
    final String record = rdfConversionUtils.convertRdfToString(rdf);

    // Perform actual normalization
    final Normalizer normalizer = new NormalizerFactory().getNormalizer();
    assertNotNull(normalizer);
    final NormalizationBatchResult result = normalizer.normalize(Collections.singletonList(record));
    assertNotNull(result);
    LOGGER.info(result.getNormalizedRecordsInEdmXml().get(0));
    //Verify conversion after normalization
    rdfConversionUtils.convertStringToRdf(result.getNormalizedRecordsInEdmXml().get(0));
  }
```

Reply:
See the attachments in jira ticket with `edm-record-date-5.xml` still get this error.
```
java.lang.NullPointerException
	at eu.europeana.normalization.model.NormalizationReport.mergeWith(NormalizationReport.java:23)
	at eu.europeana.normalization.model.NormalizationBatchResult.<init>(NormalizationBatchResult.java:30)
	at eu.europeana.normalization.NormalizerImpl.normalize(NormalizerImpl.java:51)
	at eu.europeana.normalization.NormalizationTest.testDateNormalization(NormalizationTest.java:69)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
```

Reply:
Indeed, it is in fact a different issue.
I fixed for this specific case(in the record) as well.